server.go

```diff
@@ -602,13 +606,18 @@ func (srv *Server) serveTCPListener(l Listener) error {
 		rw, err := l.AcceptWithContext(ctx)
 		if err != nil {
 			wg.Wait()
```

Comment:
`wg` must be moved under `case ErrServerClosed` -> it used to wait for go-routines that served connections.